File: src/main/java/net/ddns/rarnold/randomworkoutroutine/service/ExerciseOptionService.java
package net.ddns.rarnold.randomworkoutroutine.service;

import net.ddns.rarnold.randomworkoutroutine.model.entity.ExerciseOption;
import net.ddns.rarnold.randomworkoutroutine.repository.ExerciseOptionRepository;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class ExerciseOptionService extends ItemService<ExerciseOption> {

    private final GroupService groupService;

    public ExerciseOptionService(ExerciseOptionRepository repository, GroupService groupService) {
        super(repository);
        this.groupService = groupService;
    }

    public void delete(UUID id) {
        groupService.removeExerciseFromAll(getById(id));
        super.delete(id);
    }

    @Override
    protected ExerciseOption createItem(UUID id, String name) {
        ExerciseOption option = new ExerciseOption();
        option.setId(id);
        option.setName(name);
        return option;
    }
}

File: web/src/main/java/uk/co/rafearnold/randomworkoutroutine/web/service/ExerciseOptionService.kt
package uk.co.rafearnold.randomworkoutroutine.web.service

import org.springframework.stereotype.Service
import uk.co.rafearnold.randomworkoutroutine.web.model.entity.ExerciseOption
import uk.co.rafearnold.randomworkoutroutine.web.repository.ExerciseOptionRepository
import java.util.*

@Service
class ExerciseOptionService(repository: ExerciseOptionRepository, private val groupService: GroupService) : ItemService<ExerciseOption>(repository) {

    override fun delete(item: ExerciseOption) {
        groupService.removeExerciseFromAll(item)
        super.delete(item)
    }

    override fun createItem(id: UUID, name: String): ExerciseOption = ExerciseOption(id, name)
}
